exec/trace.go

```diff
@@ -130,12 +141,36 @@ func (t *tracer) Event(mach *sliceMachine, subject interface{}, ph string, args
 		event.Name = name.String()
 		event.Cat = "invocation"
 		key := compileKey{mach.Addr, arg.Index}
+		t.assignTid(mach, ph, t.compileEvents[key], &event)
 		t.compileEvents[key] = append(t.compileEvents[key], event)
 	default:
 		panic(fmt.Sprintf("unsupported subject type %T", subject))
 	}
 }
 
+// assignTid assigns a thread ID to event, using mach's tid pool and type of
+// event. events is the slices of existing relevant events, e.g.
+// t.taskEvents[arg].
+func (t *tracer) assignTid(mach *sliceMachine, ph string, events []traceEvent, event *traceEvent) {
+	event.Tid = 0
+	tidPool := t.machineTidPools[mach]
+	switch ph {
+	case "B":
+		event.Tid = tidPool.Acquire()
+		t.machineTidPools[mach] = tidPool
```

Comment:
Optional: I find this a little awkward. It might be less awkward to use `type tidPool struct { available []bool }`. (Also I thought `tidPool.Release` was missing its corresponding assignment, before realizing it's not. It's all perfectly reasonable and valid, but I'd guess not every Go programmer will immediately recognize not taking an address of a `map` value. But it's just a guess.)

Reply:
I did a few things to make this clearer, IMO clearer than an extra indirection to a struct, but I leave it to you to judge:

1. I made the `Release` take a pointer receiver to mirror `Acquire` to make the mutation less confusion on first glance.
2. I made the map value a pointer so that we can mutate it independent of map assignment.
3. I made the retrieval of the pool more explicit.

Let me know what you think.

Reply:
Sounds fine to me. I suspect at runtime this pointer form has the indirection. However I see no reason (like runtime performance) to avoid that in this case, so I have no complaint.